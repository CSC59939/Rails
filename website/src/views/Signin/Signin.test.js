import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Card, Icon } from 'antd';
import { Signin } from '..';
import '../../utils/tests/test.css';

configure({ adapter: new Adapter() });

describe("Signin Page", () => {
  it('Render Signin page without any error', () => { // eslint-disable-line no-undef
    expect(renderer.create(<Signin />).toJSON()).toMatchSnapshot();
  });
})
