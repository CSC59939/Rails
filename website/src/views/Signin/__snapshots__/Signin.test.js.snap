// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Page Render Signin page without any error 1`] = `
<div
  className="signin"
>
  <h1
    className="title"
  >
    Rails
  </h1>
  <h5
    className="tag-line"
  >
    Keep students on track
  </h5>
  <div
    className="ant-card signin-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Sign In
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div>
        <span
          className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <span
            className="ant-input-prefix"
          >
            <i
              className="anticon anticon-mail"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="mail"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                />
              </svg>
            </i>
          </span>
          <input
            className="ant-input ant-input-lg"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="E-Mail"
            required={true}
            size="large"
            style={null}
            type="email"
            value=""
          />
        </span>
        <span
          className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <span
            className="ant-input-prefix"
          >
            <i
              className="anticon anticon-ellipsis"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="ellipsis"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
                />
              </svg>
            </i>
          </span>
          <input
            className="ant-input ant-input-lg"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Password"
            required={true}
            size="large"
            style={null}
            type="password"
            value=""
          />
        </span>
        <button
          className="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
          onClick={[Function]}
          style={
            Object {
              "marginTop": 10,
            }
          }
          type="button"
        >
          <i
            className="anticon anticon-check-circle"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="check-circle"
              fill="currentColor"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </i>
          <p
            style={
              Object {
                "display": "inline",
                "margin": 0,
                "marginLeft": 10,
              }
            }
          >
            Sign In
          </p>
        </button>
        <a
          className="ant-btn ant-btn-lg ant-btn-block"
          href="/signup"
          onClick={[Function]}
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <p
            style={
              Object {
                "display": "inline",
                "margin": 0,
              }
            }
          >
            Don't have an account? Sign Up.
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
